src/Orders.sol

```diff
@@ -5,103 +5,110 @@ import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
-        uint256 targetChainId,
-        address tokenIn,
-        address tokenOut,
-        address recipient,
-        uint256 deadline,
-        uint256 amountIn,
-        uint256 amountOut
-    ) external payable {
+    /// @dev inputs are provided on the rollup; in exchange,
+    ///      outputs are expected to be received on the target chain(s).
+    /// @dev Fees paid to the Builders for fulfilling the Orders
+    ///      can be included within the "exchange rate" between inputs and outputs.
+    /// @dev The Builder claims the inputs from the contract by submitting `sweep` transactions within the same block.
+    /// @dev The Rollup STF MUST NOT apply `initiate` transactions to the rollup state
+    ///      UNLESS the outputs are delivered on the target chains within the same block.
+    /// @param deadline - The deadline by which the Order must be fulfilled.
+    /// @param inputs - The token amounts offered by the swapper in exchange for the outputs.
+    /// @param outputs - The token amounts that must be received on their target chain(s) in order for the Order to be executed.
+    /// @custom:reverts OrderExpired if the deadline has passed.
+    /// @custom:emits Order if the transaction mines.
+    function inititate(uint256 deadline, Input[] calldata inputs, Output[] calldata outputs) external payable {
         // check that the deadline hasn't passed
         if (block.timestamp >= deadline) revert OrderExpired();
 
-        if (tokenIn == address(0)) {
-            require(amountIn == msg.value);
-        } else {
-            IERC20(tokenIn).transferFrom(msg.sender, address(this), amountIn);
-        }
+        // transfer inputs to this contract
+        _transferInputs(inputs);
 
-        // emit the swap event
-        emit Swap(targetChainId, tokenIn, tokenOut, recipient, deadline, amountIn, amountOut);
+        // emit
+        emit Order(deadline, inputs, outputs);
+    }
+
+    /// @notice Transfer the Order inputs to this contract, where they can be collected by the Order filler.
+    function _transferInputs(Input[] calldata inputs) internal {
+        uint256 value = msg.value;
+        for (uint256 i; i < inputs.length; i++) {
+            if (inputs[i].token == address(0)) {
+                // this line should underflow if there's an attempt to spend more ETH than is attached to the transaction
+                value -= inputs[i].amount;
+            } else {
+                IERC20(inputs[i].token).transferFrom(msg.sender, address(this), inputs[i].amount);
+            }
+        }
     }
 
     /// @notice Transfer the entire balance of ERC20 tokens to the recipient.
-    /// @dev Called by the Builder within the same block as users' `swap` transactions
-    ///      to claim the amounts of `tokenIn`.
+    /// @dev Called by the Builder within the same block as users' `initiate` transactions
+    ///      to claim the `inputs`.
     /// @dev Builder MUST ensure that no other account calls `sweep` before them.
```

Comment:
this is better phrased as `MUST ensure that the sweep is atomic with the fills`

Reply:
no, I mean atomic with the fills. the builder does not need to ensure that nobody calls sweep before them, they need to ensure that the sweep (receiving payment) is atomic with the fill (filling the order)

for your issue, should probably `require(msg.sender == block.coinbase)`